test/testBrowserLanguage.js

```diff
@@ -0,0 +1,112 @@
+import browserLanguage from '../src/nginxconfig/util/browser_language';
+
+class mockLocales{
+    static languages = [];
+
+    static language = null;
+    IntlBackup = null;
+
+    setNavigatorLanguages(langs){
+        mockLocales.languages = langs;
+        return this;
+    }
+
+    setNavigatorLanguage(lang){
+        mockLocales.language = lang;
+        return this;
+    }
+
+    setDateTimeLocale(locale) {
+        const newDateTimeForMat = new Intl.DateTimeFormat(locale);
+        this.IntlBackup = Intl;
+        if (!locale){
+            // eslint-disable-next-line no-global-assign
+            Intl = undefined;
+            return;
+        }
+
+        // eslint-disable-next-line no-global-assign
+        Intl = {
+            DateTimeFormat(){
+                return newDateTimeForMat;
+            },
+        };
+        return this;
+    }
+
+    restoreDateTimeLocale(){
+        // eslint-disable-next-line no-global-assign
+        Intl = this.IntlBackup;
+        return this;
+    }
+}
+Object.defineProperty(window.navigator,'languages',{get:() => {
+    return mockLocales.languages ?? [];
+}});
+
+Object.defineProperty(window.navigator,'language',{get:() => {
+    return mockLocales.language ?? null;
+}});
+
+
+describe('#testGetBrowserLanguage', ()=> {
+    const Locale = new mockLocales();
+
+    test('#inChina', ()=> {
+        Locale.setNavigatorLanguages(['zh-CN', 'zh', 'en-US', 'en']);
+        Locale.setDateTimeLocale(undefined);
+        expect(browserLanguage()).toEqual('zhCN');
+        Locale.restoreDateTimeLocale();
+    });
+
+    test('#inHongKong',()=> {
+        Locale.setNavigatorLanguages(['zh-TW','zh','en-US','en']);
+        Locale.setDateTimeLocale(undefined);
+        expect(browserLanguage()).toEqual('zhTW');
+        Locale.restoreDateTimeLocale();
+    });
+
+    test('#inJapan',()=> {
+        Locale.setNavigatorLanguages(['ja-JP','ja']);
+        Locale.setDateTimeLocale(undefined);
+        expect(browserLanguage()).toBeFalsy();
+        Locale.restoreDateTimeLocale();
+    });
```

Comment:
Instead of creating tests based on location, we should create tests that test the potential different scenarios that the utility handles:

Assuming the available packs are `en, zhCN, zhTW`:

- Context: Exact pack match is available:

    - Test: Selects the first available exact match for language/region: `['zh-CN', 'zh', 'en-US', 'en']` => `zhCN`

    - Test: Selects the first available exact match for language: `['zh', 'en-US', 'en']` => `en`

- Context: Same language match is available:

    - Test: Selects the first available language match based on language/region: `['ja-JP', 'ja', 'en-US']` => `en`

    - Test: Selects the first available language match based on language alone: `['ja-JP', 'ja', 'zh']` => `zhCN`